### ISSUE TITLE
[ca65] cryptic error about local scope not closed

### ISSUE THREAD

**Issue by jeffythedragonslayer** [cc65/cc65#1779]
I think this error message could be improved if it also told me where the unclosed local scope first opened.  It can be hard to figure out where the mistake was made otherwise.

> Error: Local scope was not closed

**Comment by mrdudz** [comment]
Please give an example snippet that produces the message

**Comment by jeffythedragonslayer** [comment]
cryptic.s:

```
.proc unclosed

; put pages worth of code here
; put pages worth of code here
; put pages worth of code here
; put pages worth of code here
; put pages worth of code here
; put pages worth of code here
; put pages worth of code here
; put pages worth of code here
; put pages worth of code here
; put pages worth of code here
```

Running "ca65 cryptic.s" produces:
`
cryptic.s:12: Error: Local scope was not closed`

The line number is where ca65 noticed the problem.  I think there should also be a note saying line 1 is where the scope was opened.

**Comment by bbbradsmith** [comment]
In #1998 I changed the error to give the name of the scope, and in the case of proc it was able to give the line it was opened rather than the end of document line number.

Non-proc scopes don't seem to track a line number internally, as this information is apparently only attached to a label, which they don't have.

Because anonymous scopes don't appear to track any indication that they were anonymous, they may reveal their internal representation name, but this should be harmless:
```
test.s:21: Error: Local scope '$anon-SCOPE-0001' was not closed
```

**Comment by mrdudz** [comment]
fixed via #1998
